bokehjs/src/lib/core/visuals.ts

```diff
@@ -220,6 +222,15 @@ export abstract class ContextProperties {
     return value
   }
 
+  get_array(attr: string): Arrayable {
+    const array = this.cache[attr + "_array"] as Arrayable
+    if (this.all_indices != null) {
+      return map(this.all_indices, (i) => array[i])
+    } else {
+      return array
+    }
+  }
```

Comment:
Note this is a temporary solution until I get to a proper redesign of visuals (as briefly discussed in another PR).